File: src/com/colacelli/irclib/connection/listeners/OnNickChangeListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.Connection;

public interface OnNickChangeListener extends Listener {
    Type TYPE = Type.NICK;

    void onNickChange(Connection connection, User user);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/com/colacelli/irclib/connection/listeners/OnNickChangeListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection

interface OnNickChangeListener : Listener {
    fun onNickChange(connection: Connection, user: User)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.NICK
    }
}
